src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams.py

```diff
@@ -0,0 +1,425 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer, StorageAccountPreparer
+
+
+class AmsTests(ScenarioTest):
+
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer(parameter_name='storage_account_for_create')
+    def test_ams_create_show(self, resource_group, storage_account_for_create):
+        amsname = self.create_random_name(prefix='ams', length=12)
```

Comment:
Unless the AMS account name needs to be globally unique (like KeyVaults or StorageAccounts) you can get away with a static name here.

Reply:
Given that unit tests run in parallel, using the same static account name for all tests leads to name collision. Media account names should be unique by location and all test accounts are created under the same location. We could use different static account names for each test, but we believe that generating names randomly minimizes the chance of potential collisions.

Reply:
Fair enough.  Most resources only need to be unique within the RG.